packages/kbn-expandable-flyout/src/hooks/use_initialize_from_local_storage.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { EXPANDABLE_FLYOUT_LOCAL_STORAGE, PUSH_VS_OVERLAY_LOCAL_STORAGE } from '../constants';
+import { useDispatch } from '../store/redux';
+import { useExpandableFlyoutContext } from '../context';
+import { changePushVsOverlayAction } from '../store/actions';
+
+/**
+ * Hook to initialize the push vs overlay redux state from local storage
+ */
+export const useInitializeFromLocalStorage = () => {
+  const dispatch = useDispatch();
+  const { urlKey } = useExpandableFlyoutContext();
+  if (!urlKey) {
+    return;
```

Comment:
this was by design but I understand how this can be confusing. With the [latest change](https://github.com/elastic/kibana/pull/192745/commits/8e091aa08a04e2ec00357417cee519f3aa5351c0) that removes `byId` and only rely on the disable option, the memory flyouts will behave the same as non memory ones and will save the value to local storage.